app/src/main/java/com/babylon/wallet/android/utils/StringExtensions.kt

```diff
@@ -38,6 +39,12 @@ fun String.toMnemonicWords(expectedWordCount: Int): List<String> {
     }
 }
 
+fun String.sanitizedForDecimalConversion(): String {
+    val symbols = DecimalFormatSymbols.getInstance()
+    val nonDecimalCharactersInput = "[^0-9\\\\${symbols.decimalSeparator}]".toRegex()
+    return replace(nonDecimalCharactersInput, "")
+}
+
```

Comment:
I don't think you need this method any more. You can use `parseFromTextField` which returns `TextInputDecimal` which contains the `input` sanitized and the `decimal` converted.

Reply:
what about parseFromTextField throwing when I pass to it fr example `2,`. Is it expected from me to use it in `runCatching`?

Reply:
Fair point, so decimal can be nullable inside the `TextInputDecimal` class being null when the input cannot be converted to a correct decimal

Reply:
yes, that would be best. Do you want me to add that on the sargon side?